### ISSUE TITLE
attempting to ban by displayname gives unhelpful error

### ISSUE THREAD

**Issue by richvdh** [vector-im/riot-web#3459]
> Server unavailable, overloaded, or something else went wrong.

**Comment by richvdh** [comment]
The server gives a 400 with 

```
{"errcode":"M_UNKNOWN","error":"Expected UserID string to start with '@'"}
```

which is about a million times more helpful. Why are we throwing away the `error` message?

**Comment by ara4n** [comment]
because it's a generic error handler, and other errors in the API could result in spectacularly ugly "this URL failed" errors being shown to the user.

fixed in matrix-org/matrix-react-sdk@db996f6
